### PR TITLE
[codex] Harden boundary auth and path checks

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
- harden `masc_code_read` and `masc_code_write` path validation with canonicalized root containment checks
- require bearer auth on HTTP tool mutation routes instead of allowing unauthenticated same-origin fallthrough
- stop dashboard token propagation through URL and SSE query params by bootstrapping once into session storage and scrubbing `?token=`
- add targeted regression tests and include the 2026-03-31 security boundary audit report

## Root Cause
The audit found three concrete boundary failures:
- path validation relied on prefix checks, so sibling-prefix and `.worktrees/../../...` escapes were possible
- `with_tool_auth` accepted unauthenticated HTTP mutation requests when auth strictness was not enabled
- the dashboard read bearer tokens from query params and propagated them into long-lived URLs

## Impact
- closes the critical/high findings documented in #4240, #4241, and #4242
- tightens write/read confinement to the git root and worktree prefix
- makes HTTP tool mutations bearer-only
- reduces token exposure in browser-visible URLs

## Validation
- `cd dashboard && npm run typecheck`
- `cd dashboard && npm run test -- src/sse.test.ts src/api/core.test.ts src/lib/dashboard-auth.test.ts`
- `dune build --root . test/test_auth_coverage.exe test/test_tool_code_coverage.exe`
- `./_build/default/test/test_tool_code_coverage.exe test 'code_read' 3`
- `./_build/default/test/test_tool_code_coverage.exe test 'code_write' 0`
- `./_build/default/test/test_auth_coverage.exe test 'http_auth' 2`
- `./_build/default/test/test_auth_coverage.exe test 'http_auth' 3`

### CODE REVIEW

File: docs/SECURITY-BOUNDARY-AUDIT-2026-03-31.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+## Findings
+
+### Critical: code tool path validation is prefix-only and escapes repository boundaries
+
+- Code basis:
+  - [lib/tool_code.ml](/Users/dancer/me/workspace/yousleepwhen/masc-mcp/.worktrees/audit-boundaries-20260331/lib/tool_code.ml#L44)
+  - [lib/tool_code_write.ml](/Users/dancer/me/workspace/yousleepwhen/masc-mcp/.worktrees/audit-boundaries-20260331/lib/tool_code_write.ml#L25)
+- Why it fails:
+  - `validate_path` concatenates raw input and accepts it when `String.starts_with ~prefix:git_root absolute_path`.
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply docs/SECURITY-BOUNDARY-AUDIT-2026-03-31.md]: 맞습니다. 보안 감사 문서에 로컬 절대 경로가 노출되어 있습니다. 상대 경로로 변경하겠습니다.
